### PR TITLE
Avoid hanging test on M1-based Macs.

### PR DESCRIPTION
**Pull request by BjarneHerland**
On M1-based Macs, the file /bin/cat is larger than the capacity pipe-buffers, causing the test to hang in call to Popen.wait().

The solution is to add a method to determine the capacity of system pipe-buffer, then use it to limit amount of data to pipe.

### CODE REVIEW

**Review by jondequinor — COMMENTED** [review]
cat of cat is unecessary. I assume it was just to have _some_ content that exceeded 4096 bytes.

Wouldn't a better fix be to cat something > 4096 bytes from `stdin` that the test provides?

**Comment by BjarneHerland** [comment]
I agree but wanted to leave as much of the original test as possible. We could for example do something like bytearray(os.urandom(2*4096)), dump it to a tmp-file and cat it instead?

Not sure if it might be useful (apart from in this test) to know the capacity of system pipes.

**Comment by jondequinor** [comment]
> I agree but wanted to leave as much of the original test as possible. We could for example do something like bytearray(os.urandom(2*4096)), dump it to a tmp-file and cat it instead?

That's fine with me.

**Comment by BjarneHerland** [comment]
Why > 4096 btw?

**Comment by jondequinor** [comment]
Then you at least get a couple of `os.read` calls

https://github.com/equinor/ert/blob/71359963da5a034ea2db32d68c3a140c17f59bd3/res/util/subprocess.py#L36

But I don't know if that provides any value at all.

**Comment by BjarneHerland** [comment]
FWIW, I think it brings value to ensure more than one iteration.

I left the method to determine capacity of system-pipes and use it in the test. It's trivial to remove if we just decide to hardcode the data-size, but I believe it also brings value to highlight this particular limitation of Python subprocesses.

File: res/util/subprocess.py [hunk 1]
```diff
@@ -45,3 +45,16 @@ def await_process_tee(process, *out_files):
...
+# might be taken into account when using subprocesses with pipes
+def find_system_pipe_max_size():
+    from subprocess import Popen, PIPE, TimeoutExpired
+
+    p = Popen(["dd", "if=/dev/zero", "bs=1"], stdin=PIPE, stdout=PIPE)
+    try:
+        p.wait(timeout=1)
+    except:
+        p.kill()
+        return len(p.stdout.read())
```
Comment (jondequinor) [inline res/util/subprocess.py:60]: If it's not used outside testing, please move it into testing.

Comment (jondequinor) [reply res/util/subprocess.py:60]: Argument being, `res` is "API" and we want to reduce the size of it.

**Review by jondequinor — APPROVED** [review]
Golden. Thanks!

Could you make sure the `Formatting…` commit is fixupped?

**Pull request merged**